docs/smart-contracts/best-practices.md

```diff
@@ -0,0 +1,276 @@
+
+- Projects often split contract functionality into too many separate contracts—even if they share the same governance—complicating security and the overall design.
+
+## Best practices 
+
+The following list summarizes patterns, practices, and strategies for writing Pact code and delivering quality projects for the Kadena ecosystem.
+
+- Explicitly type all variables in all functions. 
+
+- Run the type checker over the code as you iterate on the implementation. 
+
+- Create unit tests for every function that fully exercise error cases. 
+
+- Be careful when using module references. 
+
+- Enforce conditions that should terminate contract execution immediately. 
+
+- Use the /local endpoint to query blockchain data. 
+
+- Only use select statements in /local queries. 
+
+- Consider what data must be on chain and any data or operations that might be better suited to off-chain handling. 
+
+- Consider using capability guards and user guards in addition to or in place of keyset guards. 
+
+- Plan for longevity and avoid defining functions that create unbounded lists and tables. 
+
+- Keep functions that share the same governance in the same module. 
+  It’s fine to create composable contracts with reusable utility functions. 
+  However, it’s generally a better practice to keep the implementation of core functions in the same module unless the functions are governed by different parties.
+
+- Design and document the capabilities your contract requires and the conditions you'll use to enforce to control access to privileged operations. 
+  Remember that capabilities can provide fine-grain access control to functions, but only if you enforce the conditions to guard them correctly.
+  If you don't plan carefully, capabilities can also leave your contract more vulnerable to attacks and misuse.
+
+- Keep in mind that the account and its associated keyset that's used to sign the transaction that deploys a contract is granted full administrative privilege over the module, including the ability to update the module and edit module tables. 
+
+## Enforcing access controls
+
+As discussed in [Capabilities](/smart-contracts/capabilities), basic and managed capabilities are critical components for controlling how permissions are granted to users of smart contracts.
+In addition, most modules define a governance capability as the module owner.
+Conceptually, capabilities aren't difficult to comprehend or implement.
+However, if they aren't used correctly, capabilities can make your contract vulnerable to unexpected behavior or to be exploited.
+
+The following examples demonstrate patterns and outcomes for governance and basic capabilities.
+
+### Governance
+
+The code defined for the GOVERNANCE capability determines the ownership of a module and its administrative privileges.
+For simplicity in code examples, the body of the GOVERNANCE capability is often set to true. 
+For example:
+
+```pact
```

Comment:
If this snippet is to be referenced later, maybe call it Figure 1. or something and reference it further down (markdown has hyperlinks so you can do this) instead of pasting it twice.